frontend/src/lib/components/CommandPalette/commandLogic.ts

```diff
@@ -69,16 +74,17 @@ const GLOBAL_COMMAND_SCOPE = 'global'
 
 function resolveCommand(
     command: Command,
-    resultsArray: CommandResult[],
+    resultsSoFar: CommandResult[],
     argument?: string,
     prefixApplied?: string
 ): void {
-    const results = Array.isArray(command.resolver) ? command.resolver : command.resolver(argument, prefixApplied)
-    resultsArray.push(
-        ...results.map((result) => {
-            return { ...result, command } as CommandResult
-        })
-    )
+    let results = command.resolver instanceof Function ? command.resolver(argument, prefixApplied) : command.resolver // run resolver or use ready-made results
+    if (!results) return // skip if no result
+    if (!Array.isArray(results)) results = [results] // work with a single result and with an array of results
+    const resultsWithCommand: CommandResult[] = results.map((result) => {
+        return { ...result, command }
+    })
+    resultsSoFar.push(...resultsWithCommand)
```

Comment:
Functions that mutate their arguments are a good source of bugs. Since this is called inside a selector, which should be more or less following functional programming principles, I'd suggest to just return `resultsWithCommand` here and push them to `resultsSoFar` inside the selector.